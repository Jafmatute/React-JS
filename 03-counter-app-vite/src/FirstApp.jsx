import PropTypes from 'prop-types';

const newMessage = 'Josue';
const getMessage = () => 'React';

const FirstApp = ({title, subTitle, name}) => {

    //if (!title) throw new Error('el titulo no existe')

    return (
        <>
            <h1>{title}</h1>
            <p> {subTitle}</p>
            <code>{name}</code>
        </>
    )
}

//PropTypes
FirstApp.propTypes = {
    subTitle: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired
}

//DefaultProps --entran primero que los PropTypes
FirstApp.defaultProps = {
    name: 'Josue Flores',
    subTitle: 123,
    //title: "No hay titulo"
}

export default FirstApp;