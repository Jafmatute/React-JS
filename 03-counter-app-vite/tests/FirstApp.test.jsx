import {render} from "@testing-library/react";
import FirstApp from "../src/FirstApp.jsx";

describe('Pruebas en el archivo <FirstApp />', ()=> {

    test('debe hacer match con el snapshot', ()=> {

        const title = 'Hola, soy Goku';

       const {container} = render(<FirstApp title={title} />)

        expect(container).toMatchSnapshot();

    })

    test('debe de mostrar el tÃ­tulo en un h1', ()=> {

        const title = 'Hola, soy Goku';

        const {container, getByText} = render(<FirstApp title={title} />)

        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');

        expect(h1.innerHTML).toBe(title);

    })
})