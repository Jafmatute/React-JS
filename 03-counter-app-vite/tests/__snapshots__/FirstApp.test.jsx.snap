// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo <FirstApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
     
    123
  </p>
  <code>
    Josue Flores
  </code>
</div>
`;
